src/hooks/useWindowSize.js

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+import useWindowEvent from '@hooks/useWindowEvent';
+
+const getWindowWidth = () => window.innerWidth;
+const getWindowHeight = () => window.innerHeight;
+
+/**
+ * @typedef  {object}           HookResponse
+ * @property {number}           width - Current window width.
+ * @property {number}           height - Current window height.
+ */
+
+/**
+ * @function
+ * @name useWindowSize
+ * @description React hook to get the current window size (width, height).
+ * This hook will listen the resize window event to always keep the size up to date.
+ * @author Felipe Nolleto Nascimento <felipenolletonascimento@gmail.com>
+ * @license https://tldrlegal.com/license/mit-license MIT
+ *
+ * @return {HookResponse} Hook response to be used within your component like props
+ */
+export default function useWindowSize() {
+  const [width, setWidth] = useState(getWindowWidth());
+  const [height, setHeight] = useState(getWindowHeight());
+
+  useWindowEvent('resize', () => {
+    setWidth(getWindowWidth());
```

Comment:
why don't you pass `width` here reading the code I assume that you should pass a value

but I get confused when I saw this line above `const [width, setWidth] = useState(getWindowWidth());`

maybe renaming ?

Reply:
This function is called when the window's size change and when that happens I must update `width` using `setWidth` (or `height` using `setHeight`).
To do that I need to pass the new width to `setWidth` and I'm using `getWindowWidth`, which get the current window size.

Reply:
go it :+1: